### PR TITLE
Erigon 3.0: Smarter full and minimal nodes

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

File: turbo/snapshotsync/snapshotsync.go [hunk 1]
```diff
@@ -126,96 +120,65 @@ func canSnapshotBePruned(name string) bool {
...
-			}
 			to, err = strconv.ParseUint(rangeNums[1], 10, 64)
 			if err != nil {
 				return nil, err
 			}
-			kind = strings.Split(name, ".")[0]
+			if stepPrune < to {
+				continue
+			}
+			blackList[name] = struct{}{}
```
Comment (AskAlexSharov) [inline turbo/snapshotsync/snapshotsync.go:151]: let's add unit-test for hits func

File: turbo/snapshotsync/snapshotsync.go [hunk 2]
```diff
@@ -106,15 +105,10 @@ func adjustBlockPrune(blocks, minBlocksToDownload uint64) uint64 {
 	if minBlocksToDownload < snaptype.Erigon2MergeLimit {
 		minBlocksToDownload = snaptype.Erigon2MergeLimit
 	}
-	if blocks < minBlocksToDownload {
+	if blocks > minBlocksToDownload {
```
Comment (AskAlexSharov) [inline turbo/snapshotsync/snapshotsync.go:108]: let's add unit-test for this func

File: turbo/cli/flags.go [hunk 3]
```diff
@@ -82,62 +82,22 @@ var (
...
-	}
-	PruneTxIndexBeforeFlag = cli.Uint64Flag{
-		Name:  "prune.t.before",
-		Usage: `Prune data before this block`,
-	}
-	PruneCallTracesBeforeFlag = cli.Uint64Flag{
-		Name:  "prune.c.before",
-		Usage: `Prune data before this block`,
-	}
-	PruneBlocksBeforeFlag = cli.Uint64Flag{
```
Comment (AskAlexSharov) [inline turbo/cli/flags.go:136]: my main question is: 
- E2 had `will external

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]
my main question is: 
`before` feature of pruning flag was added to prevent pruning of `depositContract` receipts on Mainnet. 

1 year later we added in E2 feature to not prune `depositContract`: https://github.com/ledgerwatch/erigon/pull/9123 and it made `before` feature is not useful anymore. 

But E3 doesn't store Receipts/Logs now. How will external CL work in this case? They will need re-exec old blocs to get depositContract logs? (then we need keep history `before` system contract deployed). Or we will store them? Or what?

Let's figure this out

File: turbo/cli/flags.go (hunk 3 above)
Comment (Giulio2002) [reply turbo/cli/flags.go:136]: i dont understand

File: turbo/snapshotsync/snapshotsync.go (hunk 1 above)
Comment (Giulio2002) [reply turbo/snapshotsync/snapshotsync.go:151]: there are, i just simlified the func

File: turbo/snapshotsync/snapshotsync.go (hunk 2 above)
Comment (Giulio2002) [reply turbo/snapshotsync/snapshotsync.go:108]: there unit tests for blacklist

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**